tape/src/main/java/com/squareup/tape2/BlockingFileQueue.java

```diff
@@ -0,0 +1,407 @@
+package com.squareup.tape2;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * This is a simple and unbounded @{@link BlockingQueue} implementation as a wrapper around a @{@link QueueFile}.
+ * Thread safety is implemented using a single lock around all operations to the backing @{@link QueueFile}.
+ */
+public class BlockingFileQueue implements BlockingQueue<byte[]> {
```

Comment:
Rather than `byte[]` - what do you think about using a converter interface https://github.com/square/tape/blob/master/tape/src/main/java/com/squareup/tape2/ObjectQueue.java#L86-L92 to allow custom types to be used?